### ISSUE TITLE
Request: Add AIS sentences

### ISSUE THREAD

**Issue by emilecantin** [SignalK/signalk-to-nmea0183#47]
We do the reverse in the parser module, why not here too? Also, I'd actually need it for my use-case.

**Comment by sbender9** [comment]
It should be here, just need someone to implement it.

What's your source for AIS data?

**Comment by emilecantin** [comment]
I started looking at it, but it's a bit more complex than I anticipated, because we currently filter on the "self" values.

I'm actually getting AIS data from an NMEA0183 source (Moitessier HAT on a Raspberry Pi running OpenPlotter), but I'm sending all my data to another Raspberry Pi running OpenPlotter through SignalK. Everything works fine, except OpenCPN doesn't see any AIS data.

My end goal is to have one "headless" Pi running all my sensors, feeding another Pi acting as a chartplotter through SignalK. I'm pretty close, the only thing missing is AIS.

Of course, I could just connect directly to kplex on the other Pi to subscribe directly to the source NMEA data, but where's the fun in that?

**Comment by itemir** [comment]
+1 for this request. Use-case is to use the existing Signal K server on a Raspberry Pi that is pulling AIS data from a USB connected AIS transceiver for tablets and OpenCPN over WiFi port 10110.

Cool plugin btw, apparently I have had this for a while but didn't go into specifics.

**Comment by tkurki** [comment]
@itemir your AIS transceiver is probably already outputting nmea0183 ais data, you don’t need to do the conversion from SK.

**Comment by itemir** [comment]
It is but it is doing so over USB, which is feeding into Signal K. Goal is to make it available over WiFi. I can feed into Signal K and a separate wireless solution with something like gpsd but it complicates things. Would be cool to give it out from SK.

**Comment by tkurki** [comment]
All SK server's data from nmea0183 inputs is already available over tcp on port 10110 - it does not have to be converted first to Signal K and then back to nmea0183, if it is coming from an nmea0183 source.

**Comment by itemir** [comment]
As I mentioned above, AIS comes via USB and it is not showing up on port 10110.

**Comment by tkurki** [comment]
I don't doubt your word on ais on port 10110, but the source being USB is not the reason, trust me. But this is getting beyond this issue, care to come to Slack if you want to track this down further? Link to join is at http://signalk.org/

**Comment by itemir** [comment]
@julled what do you mean by that? We have been using AIS through Signal K. What exactly are you trying to achieve?

**Comment by itemir** [comment]
Yes that is correct, you should see it without problems. What is the source of those AIS messages? Is that an N2k device? And did you try it and is it not working?

Make sure you have the `signalk-n2kais-to-nmea0183` plugin.

**Comment by tkurki** [comment]
I am closing this issue as it is now a mixture of several different things. 

If somebody really needs conversion from sk to nmea0183 ais then we can open a new issue, this issue does not really provide any value.

**Comment by Nosen92** [comment]
Check out [signalk-vessels-to-ais](https://github.com/KEGustafsson/signalk-vessels-to-ais) for this functionality.
